Automattic/jetpack
# Boost: Improve active module list in speed score requests

Speed score requests store which modules are active at the time of the request. This is to be able to compare it with the score of `no-boost` modules active and figuring out the impact of boost modules. Previously, it was just looking at what modules are enabled. But, this may not always be enough.

Example:
- Critical CSS and Cloud CSS need to generate the CSS before it can be useful.
- Cache engine needs to load successfully for it to help optimize the site.

## Proposed changes:

* In Speed Score requests, keep record of the modules that are actually working, just being enabled is not enough.

### Other information:


## Jetpack product discussion
None

## Does this pull request change what data or activity we track or use?
None

## Testing instructions:
* Enable cache.
* Rig the `advanced-cache.php` file so the caching engine doesn't load.
* Run a speed score and check if `page_cache` is stored as an `active_module` in [`Speed_Score_Request`](https://github.com/Automattic/jetpack/blob/9f7b71faae30adb636c47d29aacd8be4fc44b679/projects/packages/boost-speed-score/src/class-speed-score-request.php#L17).
